docs/howtos/http-requests.md

```diff
@@ -0,0 +1,210 @@
+# How to make HTTP requests using `PyScript`, in pure Python
+
+[Pyodide](https://pyodide.org), the runtime that underlies `PyScript`, does not have the `requests` module
+(or other similar modules) available by default, which are traditionally used to make HTTP requests in Python.
+However, it is possible to make HTTP requests in Pyodide using the modern `JavaScript` `fetch` API
+([docs](https://developer.mozilla.org/en-US/docs/Web/API/fetch)). This example shows how to make common HTTP request
+(GET, POST, PUT, DELETE) to an API, using only python code! We will use asynchronous functions with
+async/await syntax, as concurrent code is preferred for HTTP requests.
+
+The purpose of this guide is not to teach the basics of HTTP requests, but to show how to make them
+from `PyScript` using python, since currently, the common tools such as `requests` and `httpx` are not available.
+
+## Fetch
+
+The `fetch` API is a modern way to make HTTP requests. It is available in all modern browsers, and in Pyodide.
+
+Although there are two ways to use `fetch`, 1) using `JavaScript` from `PyScript`, and 2) using Pyodide's python wrapper,
+`Pyodide.http.pyfetch`, this example will only show how to use the python wrapper. Still, the
+[fetch documentation](https://developer.mozilla.org/en-US/docs/Web/API/fetch#parameters) is a useful reference, as its
+parameters can be called from python using the `pyfetch` wrapper.
+
+## Pyodide.http, pyfetch, and FetchResponse
+
+[Pyodide.http module](https://pyodide.org/en/stable/usage/api/python-api/http.html#module-pyodide.http) is a python API
+for dealing with HTTP requests. It provides the `pyfetch` function as a wrapper for the `fetch` API,
+which returns a `FetchResponse` object whenever a request is made. Extra keyword arguments can be passed to `pyfetch`
+which will be passed to the `fetch` API.
+
+The returned object `FetchResponse` has familiar methods and properties
+for dealing with the response, such as `json()` or `status`. See the
+[FetchResponse documentation](https://pyodide.org/en/stable/usage/api/python-api/http.html#pyodide.http.FetchResponse)
+for more information.
+
+# Example
+We will make async HTTP requests to [JSONPlaceholder](https://jsonplaceholder.typicode.com/) fake API using `pyfetch`.
```

Comment:
We will make async HTTP requests to [JSONPlaceholder](https://jsonplaceholder.typicode.com/)'s fake API using `pyfetch`.

Reply:
Clarified what I meant